### PR TITLE
Rename HardExampleAlgorithm to HardExampleMining

### PR DESCRIPTION
**Pull request by TymonXie**
1.rename HardExampleAlgorithm to HardExampleMining
2.regenerate zz_generated.deepcopy.go

### CODE REVIEW

File: build/crds/neptune/jointinferenceservice_v1alpha1.yaml [hunk 1]
```diff
@@ -1,194 +1,194 @@
-apiVersion: apiextensions.k8s.io/v1
```
Comment (llhuii) [inline build/crds/neptune/jointinferenceservice_v1alpha1.yaml:1]: Format of this file should be unchanged

**Review by llhuii — COMMENTED** [review]
joint inference crd sample should be updated too.

File: build/crds/neptune/jointinferenceservice_v1alpha1.yaml (hunk 1 above)
Comment (TymonXie) [reply build/crds/neptune/jointinferenceservice_v1alpha1.yaml]: Has fixed, thx!

File: pkg/apis/neptune/v1alpha1/jointinferenceservice_types.go [hunk 2]
```diff
@@ -27,7 +27,7 @@ type JointInferenceServiceSpec struct {
 type EdgeWorker struct {
 	Model                SmallModel           `json:"model"`
 	NodeName             string               `json:"nodeName"`
-	HardExampleAlgorithm HardExampleAlgorithm `json:"hardExampleAlgorithm"`
+	HardExampleMining HardExampleMining `json:"hardExampleAlgorithm"`
```
Comment (llhuii) [inline pkg/apis/neptune/v1alpha1/jointinferenceservice_types.go:30]: json:hardExampleAlgorithm needs to be updated

**Review by llhuii — DISMISSED** [review]

**Comment by TymonXie** [comment]
joint inference crd sample has been updated.

File: pkg/apis/neptune/v1alpha1/jointinferenceservice_types.go (hunk 2 above)
Comment (TymonXie) [reply pkg/apis/neptune/v1alpha1/jointinferenceservice_types.go]: has been updated

**Review by llhuii — APPROVED** [review]

**Pull request merged**